extensions/binding/org.eclipse.smarthome.binding.fsinternetradio/src/main/java/org/eclipse/smarthome/binding/fsinternetradio/internal/FSInternetRadioHandlerFactory.java

```diff
@@ -35,13 +35,6 @@ public boolean supportsThingType(ThingTypeUID thingTypeUID) {
 
     @Override
     protected ThingHandler createHandler(Thing thing) {
-
-        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
-
-        if (thingTypeUID.equals(THING_TYPE_RADIO)) {
-            return new FSInternetRadioHandler(thing);
-        }
-
-        return null;
+        return new FSInternetRadioHandler(thing);
```

Comment:
I removed that check from here because  when the ThingManager class is trying to register a Handler it calls the `findThingHandlerFactory(ThingTypeUID thingTypeUID)` method and the registerHandler method is never called if the factory does not support the given thingTypeUID. I tried to test it and that `return null;` line can never be covered.